ptest/cases/autonomous_mission_manager_case.py

```diff
@@ -0,0 +1,112 @@
+from .base import MissionCase
+import time
+from psim.sims import DualAttitudeOrbitGnc
+import lin
+from .utils import str_to_val
+
+class AutonomousMissionManagerCase(MissionCase):
+
+    def state_check(self, satellite, designaton):
+        satellite_state = satellite.read_state("pan.state")
+        if(satellite_state == "3"):
+            self.logger.put(designaton + " is in standby. Ending mission." )
+            return False
+        elif(satellite_state == "10"):
+            self.logger.put(designaton + " is in state safehold. Ending mission.")
+            return False
+        return True
+    
+    def mission_conditions_met(self):
+        #check for operating leader state
+        leader_state_functional = self.state_check(self.leader, "Leader")
+        follower_state_functional = self.state_check(self.follower, "Follower")
+        if not leader_state_functional:
+            if follower_state_functional:
+                self.follower.write_state("pan.state", 3)
+            else:
+                self.follower.write_state("pan_state", 10)
+            return False
+
+        #check faulting state for follower state
+        if not follower_state_functional:
+            self.leader.write_state("pan.state", 3)
+            return False
+
+        #check time since last comms
+        leader_time_since_comms = time.time() - self.leader_time_last_comms 
+        if(leader_time_since_comms > self.comms_time_threshold):
+            self.logger.put("Leader is experiencing comms blackout. Ending mission." + str(leader_time_since_comms))
+            return False
+        follower_time_since_comms = time.time() - self.follower_time_last_comms 
+        if(follower_time_since_comms > self.comms_time_threshold):
+            self.logger.put("Follower is experiencing comms blackout. Ending mission." + str(follower_time_since_comms))
+            return False
+
+
+        #TODO notify if close to docking
+        #TODO exit gracefully if mission ended
+
+        return True
+    
+    def propagate_orbits(self, vals):
+        return vals
+
+
+    @property
+    def sim_configs(self):
+        configs = ["truth/ci", "truth/base"]
+        configs += ["sensors/base"]
+        return configs
+
+    @property
+    def sim_model(self):
+        return DualAttitudeOrbitGnc
+
+    @property
+    def sim_mapping(self):
+        return "ci_mapping.json"
+        
+    @property
+    def sim_duration(self):
+        return float("inf")
+
+    def setup_post_bootsetup_follower(self):
+        self.using_radios = self.radio_follower != None #assumes only using one physical radio
+        self.leader = self.flight_controller_leader
+        self.follower = self.radio_follower if self.using_radios else self.flight_controller_follower
+
+        self.leader_time_last_comms = time.time()
+        self.follower_time_last_comms = time.time()
+        self.comms_time_threshold = 60*5*1000000000 #currently 5 minutes for testing
```

Comment:
```suggestion
        self.comms_time_threshold = 60*5 #currently 5 minutes for testing
```
time.time() returns a float in [seconds](https://docs.python.org/3/library/time.html#time.time)